### PR TITLE
[Inductor] Skip nonfinite preservation for safe scaled softmax

### PR DESCRIPTION
**Pull request by Yuxingwang-intel**
Skip the scaled softmax nonfinite-semantics preservation logic for static safe scale factors.

PR https://github.com/pytorch/pytorch/pull/184046 added logic in Inductor to preserve the original scaled subtraction behavior for nonfinite scaled-softmax inputs, so that Inductor does not turn eager NaNs into finite softmax results. However, for static positive scales <= 1, multiplying finite inputs by the scale will not introduce overflow. In those cases, the additional preservation logic is unnecessary overhead and can also block downstream fusion patterns or cause performance regressions on benchmarks.
This PR keeps the existing nonfinite-semantics preserving path for unsafe or dynamic scales but directly returns the stable scaled-softmax form for compile-time safe scales.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
@claude explain those CI failures, check if they are related to this PR, and review the changes

**Comment by Yuxingwang-intel** [comment]
Hi @eellison, could you please help review this PR? Thanks.

**Comment by Yuxingwang-intel** [comment]
Hi @Valentine233, could you please help review this PR? Thanks.

**Review by Valentine233 — COMMENTED** [review]
Thanks! Please modify according to Claude's review. The 2nd and 3rd points make sense to me. In addition, please fix the CI failures.

**Comment by eellison** [comment]
adding @frgossen who reviewed that pr

**Comment by Yuxingwang-intel** [comment]
> Thanks! Please modify according to Claude's review. The 2nd and 3rd points make sense to me. In addition, please fix the CI failures.

Thanks! Done. The CI failure seems unrelated to my changes.

**Review by Valentine233 — APPROVED** [review]

**Comment by Yuxingwang-intel** [comment]
@claude explain those CI failures, check if they are related to this PR.

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by Yuxingwang-intel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by Yuxingwang-intel** [comment]
The failures do not seem to be related to my commit. This PR only changes the Inductor scaled-softmax pattern in joint_graph.py.
